### PR TITLE
add helpers to capi sdk for creating aws role identity

### PR DESCRIPTION
**Pull request by Rohitrajak1807**
Ticket link: https://platform9.atlassian.net/browse/EMP-2211

### CODE REVIEW

File: infrastructure/infrastructure.go [hunk 1]
```diff
@@ -9,16 +12,24 @@ type InfraProvider interface {
...
 	CreateInfraMachineTemplate(ctx context.Context, input CreateInfraMachineTemplateInput) error
 
 	DeleteInfraCluster(ctx context.Context, input DeleteInfraClusterInput) error
-	DeleteInfraStaticIdentity(ctx context.Context, input DeleteInfraClusterStaticIdentityInput) error
+	DeleteInfraStaticIdentity(ctx context.Context, input DeleteInfraClusterIdentityInput) error
 	DeleteInfraMachine(ctx context.Context, input DeleteInfraMachineInput) error
 	DeleteInfraMachineTemplate(ctx context.Context, input DeleteInfraMachineTemplateInput) error
+
+	CreateClusterRoleIdentity(ctx context.Context, input CreateInfraClusterIdentityInput) error
+	GetClusterRoleIdentity(ctx context.Context, input GetClusterRoleIdentityInput) error
```
Comment (OmkarDeshpande7) [inline infrastructure/infrastructure.go:25]:
```suggestion
	GetClusterRoleIdentity(ctx context.Context, input GetClusterIdentityInput) error
```
make this also generic

**Review by OmkarDeshpande7 — COMMENTED** [review]
one small comment. please resolve conflicts. Looks good otherwise

File: infrastructure/infrastructure.go (hunk 1 above)
Comment (Rohitrajak1807) [reply infrastructure/infrastructure.go]: done

**Comment by Rohitrajak1807** [comment]
@OmkarDeshpande7 it's done
please review again. Feel free to hit merge

**Review by OmkarDeshpande7 — APPROVED** [review]

**Pull request merged**